### ISSUE TITLE
Buildx multiarch image date and history not shown

### ISSUE THREAD

**Issue by GoryMoon** [Joxit/docker-registry-ui#309]
Hi, I use this docker registry UI and I have an issue...

## Bug description

Moved from the comment in https://github.com/Joxit/docker-registry-ui/issues/308#issuecomment-1549366336 to it's own issue.

Buildx added `provenance` as default in [v0.10](https://github.com/docker/buildx/releases/tag/v0.10.0), this changed it to use `OCI` tags to be compliant.
To be able to show history you need to disable this by adding `--provenance=false` to the build command.

I build my images more or less like this:
```sh
docker buildx build --platform linux/amd64,linux/arm64 --tag <image name> --output=type=registry .
```

I don't see how the registry UI couldn't support this for multiarch builds as when looking at the API response it's very similar to how the docker manifest is.
This is what is returned with the manifest from a tag.
> E.g. `/v2/<image name>/manifests/latest`:
<details>
<summary>Click to expand tag manifest </summary>

```json
{
    "mediaType": "application/vnd.oci.image.index.v1+json",
    "schemaVersion": 2,
    "manifests": [
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "digest": "sha256:9f38533c3ac11fe5b6d9c08aa5f683887159ecbc01ee65635b2fb3ab6de57ab0",
            "size": 2761,
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "digest": "sha256:caac61937ce912af68dd7b09ea16be9e15211f7319d800913d19a80b73b4a298",
            "size": 2761,
            "platform": {
                "architecture": "arm64",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "digest": "sha256:d388def5035290ee739d973cc9e8549e237f423de26dee55ea4949a40d01da65",
            "size": 566,
            "annotations": {
                "vnd.docker.reference.digest": "sha256:9f38533c3ac11fe5b6d9c08aa5f683887159ecbc01ee65635b2fb3ab6de57ab0",
                "vnd.docker.reference.type": "attestation-manifest"
            },
            "platform": {
                "architecture": "unknown",
                "os": "unknown"
            }
        },
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "digest": "sha256:6f5e15228ac5ad5dacfbbfc97df6564a9ab1cd0adcdb197b00a627cbe3b7cc51",
            "size": 566,
            "annotations": {
                "vnd.docker.reference.digest": "sha256:caac61937ce912af68dd7b09ea16be9e15211f7319d800913d19a80b73b4a298",
                "vnd.docker.reference.type": "attestation-manifest"
            },
            "platform": {
                "architecture": "unknown",
                "os": "unknown"
            }
        }
    ]
}
```

</details>

You can use the digest of the previous manifest like when you are using `application/vnd.docker.distribution.manifest.list.v2+json` to get the actual image here:

https://github.com/Joxit/docker-registry-ui/blob/bc80050a4488b94cafe0595f7969dff13070c0c2/src/scripts/docker-image.js#L102-L111

Should probably only do this if the response from the tag manifest response is of the `application/vnd.oci.image.manifest.v1+json` type.

When using the digest from the first manifest this is what's returned.
And using that `config` blob returns the history as the current system expects.
> E.g. `/v2/<image name>/manifests/sha256:9f38533c3ac11fe5b6d9c08aa5f683887159ecbc01ee65635b2fb3ab6de57ab0`:
<details>
<summary>Click to expand digest manifest</summary>

```json
{
    "mediaType": "application/vnd.oci.image.manifest.v1+json",
    "schemaVersion": 2,
    "config": {
        "mediaType": "application/vnd.oci.image.config.v1+json",
        "digest": "sha256:8f2e7c70dac05c95bd5a522cf4a059857fc8c997ad1cdf9e2b3eff9a37fe9c4c",
        "size": 9253
    },
    "layers": [
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:1efc276f4ff952c055dea726cfc96ec6a4fdb8b62d9eed816bd2b788f2860ad7",
            "size": 31366757
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:a2f2f93da48276873890ac821b3c991d53a7e864791aaf82c39b7863c908b93b",
            "size": 1582262
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:12cca292b13cb58fadde25af113ddc4ac3b0c5e39ab3f1290a6ba62ec8237afd",
            "size": 212
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:d73cf48caaac2e45ad76a2a9eb3b311d0e4eb1d804e3d2b9cf075a1fa31e6f92",
            "size": 46042445
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:e832be86182094add78a1690b0fbd8818e40710a53d43171db4b271d43d46158",
            "size": 5484
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:871f7cebf585d387fde9f3fd270668065701606577a1008ed3b3acfac8274cc4",
            "size": 18460921
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:207983e6a84c88158ffa317f4341aa449fe063b44834682569d7b2506c9f2437",
            "size": 859851653
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:1d99b33ce83abb0fd774713bf9dd14c29710d54f969b57056a35f5a9b332054e",
            "size": 179
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:ca5cba012caf700f0fb5ddb9a7904eae14744e4ab44594c1b482ecbad77f8d1b",
            "size": 3057
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:39a7c4fc07c05d4870b4d95a7a0015ab56c4e07e516cf549a54bf3e9fd8ff73a",
            "size": 212
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:401b2e2746eb54009b1f49b321161cfc251c1c8d12444b0eeb463a9b4f1902ea",
            "size": 329
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:4eafb804449bab1cdde11fb488b0a7a1f59f89bf24a530463a1cb91c07eecf8a",
            "size": 5323
        },
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
            "digest": "sha256:0f7e12b417c3dbebe0a2ba16556075c1b64823b25ba115c43ec14f494d6c64b4",
            "size": 390
        }
    ]
}
```

</details>

#### For service bug, steps to reproduce the behavior:
My docker-compose file
```yml
version: '3'

services:
  registry:
    container_name: registry
    image: registry:2
    volumes:
      - ./registry/data:/var/lib/registry
      - ./registry/config.yml:/etc/docker/registry/config.yml
    environment:
      - REGISTRY_STORAGE_DELETE_ENABLED=true
      - REGISTRY_HTTP_SECRET=<secret>
    restart: unless-stopped
    labels:
      traefik.enable: true
      traefik.http.routers.registry.entrypoints: https
      traefik.http.routers.registry.rule: Host(`registry.example.com`)
      traefik.http.services.registry.loadbalancer.server.port: 5000
    networks:
      - traefik-proxy

  registry-ui:
    container_name: registry-ui
    image: joxit/docker-registry-ui:latest
    environment:
      - REGISTRY_TITLE=Registry Repo
      - REGISTRY_URL=https://registry.example.com
      - SHOW_CONTENT_DIGEST=true
      - SINGLE_REGISTRY=true
      - DELETE_IMAGES=true
      - CATALOG_ELEMENTS_LIMIT=1000
    restart: unless-stopped
    depends_on:
      - registry
    labels:
      traefik.enable: true
      traefik.http.routers.registry-ui.entrypoints: https
      traefik.http.routers.registry-ui.rule: Host(`registry-ui.example.com`)
      traefik.http.services.registry-ui.loadbalancer.server.port: 80
    networks:
      - traefik-proxy
```

My private docker registry configuration
```yml
version: 0.1
log:
  fields:
    service: registry
storage:
  delete:
    enabled: true
  cache:
    blobdescriptor: inmemory
  filesystem:
    rootdirectory: /var/lib/registry
http:
  addr: :5000
  headers:
    X-Content-Type-Options: [nosniff]
    Access-Control-Allow-Origin: ['*']
    Access-Control-Allow-Headers: ['Authorization', 'Accept']
    Access-Control-Allow-Methods: ['HEAD', 'GET', 'OPTIONS', 'DELETE']
    Access-Control-Max-Age: [1728000]
    Access-Control-Expose-Headers: ['Docker-Content-Digest']

```

## Expected behavior

To be able to view history on multiarch OCI images.

## System information

- OS: Debian 11
- Docker registry UI:
    - Version: 2.4.1
    - Server: docker
    - Docker version: 23.0.5
    - Docker registry version: 2.8.2
    - Docker registry ui tag: latest
    - OS/Arch: linux/arm64
    - Tools: docker-compose

**Comment by Joxit** [comment]
Hi an thank you for using my project and for your detailed issue, I really appreciate this :smile: 

First of all, you can check this issue and comment from 2022, it will exaplain the reason why OCI images are not supported in the UI: https://github.com/Joxit/docker-registry-ui/issues/227#issuecomment-1069698250 
The used tool was `buildctl` and not `buildx` but IMO they should work in the same way ?

As I can see, the most important part is missing from your issue, the part where I fetch the blob of the manifest. This is the response I use to retrieve the history and image date. Since this endpoint did not work back in 2022 on OCI images, I decided to display an error message to warn user.

---

I've tried your configuration and it seems that responses are totally different than the ones I had with `buildctl` in the previous issue. Can you tell me if you use curl to the endpoint bellow you will receive a JSON like mine ?

```
GET /v2/<image name>/blobs/<config.digest string>

Accept: application/vnd.docker.distribution.manifest.v2+json, application/vnd.oci.image.manifest.v1+json
```

<details>
  <summary>Click to expand Blob Result</summary>

```json
{
  "architecture": "amd64",
  "config": {
    "ExposedPorts": {
      "80/tcp": {}
    },
    "Env": [
      "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
      "NGINX_VERSION=1.21.6",
      "NJS_VERSION=0.7.2",
      "PKG_RELEASE=1",
      "NGINX_PROXY_HEADER_Host=$http_host",
      "NGINX_LISTEN_PORT=80"
    ],
    "Entrypoint": [
      "/docker-entrypoint.sh"
    ],
    "Cmd": [
      "nginx",
      "-g",
      "daemon off;"
    ],
    "WorkingDir": "/usr/share/nginx/html/",
    "Labels": {
      "maintainer": "Jones MAGLOIRE @Joxit"
    },
    "StopSignal": "SIGQUIT",
    "OnBuild": null
  },
  "created": "2022-03-13T08:13:28.458764628+01:00",
  "history": [
    {
      "created": "2021-11-24T20:19:40.199700946Z",
      "created_by": "/bin/sh -c #(nop) ADD file:9233f6f2237d79659a9521f7e390df217cec49f1a8aa3a12147bbca1956acdb9 in / "
    },
    {
      "created": "2021-11-24T20:19:40.483367546Z",
      "created_by": "/bin/sh -c #(nop)  CMD [\"/bin/sh\"]",
      "empty_layer": true
    },
    {
      "created": "2021-12-29T19:29:04.16538557Z",
      "created_by": "/bin/sh -c #(nop)  LABEL maintainer=NGINX Docker Maintainers <docker-maint@nginx.com>",
      "empty_layer": true
    },
    {
      "created": "2022-01-25T20:38:57.564847523Z",
      "created_by": "/bin/sh -c #(nop)  ENV NGINX_VERSION=1.21.6",
      "empty_layer": true
    },
    {
      "created": "2022-01-25T20:38:57.780944464Z",
      "created_by": "/bin/sh -c #(nop)  ENV NJS_VERSION=0.7.2",
      "empty_layer": true
    },
    {
      "created": "2022-01-25T20:38:58.001318895Z",
      "created_by": "/bin/sh -c #(nop)  ENV PKG_RELEASE=1",
      "empty_layer": true
    },
    {
      "created": "2022-01-25T20:39:07.536319062Z",
      "created_by": "/bin/sh -c set -x  ...."
    },
    {
      "created": "2022-01-25T20:39:08.151512909Z",
      "created_by": "/bin/sh -c #(nop) COPY file:65504f71f5855ca017fb64d502ce873a31b2e0decd75297a8fb0a287f97acf92 in / "
    },
    {
      "created": "2022-01-25T20:39:08.618949282Z",
      "created_by": "/bin/sh -c #(nop) COPY file:0b866ff3fc1ef5b03c4e6c8c513ae014f691fb05d530257dfffd07035c1b75da in /docker-entrypoint.d "
    },
    {
      "created": "2022-01-25T20:39:09.304128562Z",
      "created_by": "/bin/sh -c #(nop) COPY file:0fd5fca330dcd6a7de297435e32af634f29f7132ed0550d342cad9fd20158258 in /docker-entrypoint.d "
    },
    {
      "created": "2022-01-25T20:39:09.813218835Z",
      "created_by": "/bin/sh -c #(nop) COPY file:09a214a3e07c919af2fb2d7c749ccbc446b8c10eb217366e5a65640ee9edcc25 in /docker-entrypoint.d "
    },
    {
      "created": "2022-01-25T20:39:10.15290576Z",
      "created_by": "/bin/sh -c #(nop)  ENTRYPOINT [\"/docker-entrypoint.sh\"]",
      "empty_layer": true
    },
    {
      "created": "2022-01-25T20:39:10.498199739Z",
      "created_by": "/bin/sh -c #(nop)  EXPOSE 80",
      "empty_layer": true
    },
    {
      "created": "2022-01-25T20:39:10.860347544Z",
      "created_by": "/bin/sh -c #(nop)  STOPSIGNAL SIGQUIT",
      "empty_layer": true
    },
    {
      "created": "2022-01-25T20:39:11.21594189Z",
      "created_by": "/bin/sh -c #(nop)  CMD [\"nginx\" \"-g\" \"daemon off;\"]",
      "empty_layer": true
    },
    {
      "created": "2022-03-13T08:13:28.237041716+01:00",
      "created_by": "LABEL maintainer=Jones MAGLOIRE @Joxit",
      "comment": "buildkit.dockerfile.v0",
      "empty_layer": true
    },
    {
      "created": "2022-03-13T08:13:28.237041716+01:00",
      "created_by": "WORKDIR /usr/share/nginx/html/",
      "comment": "buildkit.dockerfile.v0"
    },
    {
      "created": "2022-03-13T08:13:28.237041716+01:00",
      "created_by": "ENV NGINX_PROXY_HEADER_Host=$http_host",
      "comment": "buildkit.dockerfile.v0",
      "empty_layer": true
    },
    {
      "created": "2022-03-13T08:13:28.237041716+01:00",
      "created_by": "ENV NGINX_LISTEN_PORT=80",
      "comment": "buildkit.dockerfile.v0",
      "empty_layer": true
    },
    {
      "created": "2022-03-13T08:13:28.292012782+01:00",
      "created_by": "COPY nginx/default.conf /etc/nginx/conf.d/default.conf # buildkit",
      "comment": "buildkit.dockerfile.v0"
    },
    {
      "created": "2022-03-13T08:13:28.309117311+01:00",
      "created_by": "COPY bin/entrypoint /docker-entrypoint.d/90-docker-registry-ui.sh # buildkit",
      "comment": "buildkit.dockerfile.v0"
    },
    {
      "created": "2022-03-13T08:13:28.336973324+01:00",
      "created_by": "COPY dist/ /usr/share/nginx/html/ # buildkit",
      "comment": "buildkit.dockerfile.v0"
    },
    {
      "created": "2022-03-13T08:13:28.35325107+01:00",
      "created_by": "COPY favicon.ico /usr/share/nginx/html/ # buildkit",
      "comment": "buildkit.dockerfile.v0"
    },
    {
      "created": "2022-03-13T08:13:28.458764628+01:00",
      "created_by": "RUN /bin/sh -c chown -R nginx:nginx /etc/nginx/ /usr/share/nginx/html/ /var/cache/nginx # buildkit",
      "comment": "buildkit.dockerfile.v0"
    }
  ],
  "moby.buildkit.buildinfo.v1": "eyJmcm9udGVuZCI6ImRvY2tlcmZpbGUudjAiLCJzb3VyY2VzIjpbeyJ0eXBlIjoiZG9ja2VyLWltYWdlIiwicmVmIjoiZG9ja2VyLmlvL2xpYnJhcnkvbmdpbng6YWxwaW5lIiwicGluIjoic2hhMjU2OmRhOWM5NGJlYzFkYTgyOWViZDUyNDMxYTg0NTAyZWM0NzFjOGU1NDhmZmIyY2VkYmYzNjI2MGZkOWJkMWQ0ZDMifV19",
  "os": "linux",
  "rootfs": {
    "type": "layers",
    "diff_ids": [
      "sha256:8d3ac3489996423f53d6087c81180006263b79f206d3fdec9e66f0e27ceb8759",
      "sha256:67bae81de3dc5ccf5101dd4973e8e4b10ee3d5681feb9c30119e6acee1e12dd5",
      "sha256:89f4d03665cee1ca943d0b55c086ea7577f4658c2cabe69eaf640f1dc47129f9",
      "sha256:318191938fd7801d80ad617482e4312f0ced71e6c1d4bcb02e12d90a9d65eb11",
      "sha256:a770f8eba3cb3840d7579e2c4925c9fa275813e8719fe5435905b887a9ee7674",
      "sha256:6fda88393b8b02872c87c6626807e7bfac1f49e63f47a5ddba0cd11fae99b575",
      "sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef",
      "sha256:16eb951e9ea9c0ab11e5af087c762528f85e6b216a7c56ef003805439ea5f90f",
      "sha256:1a1db89726971b65895b495442ed10b5df0d4c7e90cbb943d28b4fce770257cb",
      "sha256:5c8eff90fb43c7dbe872fdf130dc51ab8828cd9d461af9944c3ad8ba4286301a",
      "sha256:18749a2b685b525bef320c94d198d7b40fc83a08bc1d951cc1e3d3b121e2ff92",
      "sha256:254c4dc7aefcff4cca8b682e73b9297707f6d8874626a167a5cb8ac23149d12e"
    ]
  }
}
```

</details>

**Comment by GoryMoon** [comment]
Buildx extends the capabilities from BuildKit so they should work the same way, I guess, but `buildx` probably has some different options applied/additional work done on the image.

I've setup a complete separate image and repo to show all stages.
The command I had at the top is the command I used to build the image in these examples.
The curl commands include the required headers to fetch the different urls.
<hr/>

### Catalog list:
`curl -X GET https://<repo>/v2/_catalog`
<details>
<summary>Click to show Catalog</summary>

```json
{
  "repositories": [
    "gorymoon/sample"
  ]
}
```

</details>
<hr/>

### Tag list
`curl -X GET https://<repo>/v2/gorymoon/sample/tags/list`
<details>
<summary>Click to show Tags</summary>

```json
{
  "name": "gorymoon/sample",
  "tags": [
    "latest"
  ]
}
```

</details>
<hr/>

### Tag index:
`curl -X GET -H 'Accept: application/vnd.oci.image.index.v1+json' https://<repo>/v2/gorymoon/sample/manifests/latest`
<details>
<summary>Click to show Tag Index</summary>

```json
{
  "mediaType": "application/vnd.oci.image.index.v1+json",
  "schemaVersion": 2,
  "manifests": [
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:385c5ec07e72af7f5acb6173b5dc704f05ffc33ec14c97ed961d62d2a5cceba7",
      "size": 1616,
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:4c48a7c712ff3446d982c7131ecf87b4e17f25495b8ac27789e5aedfd28762ae",
      "size": 1616,
      "platform": {
        "architecture": "arm64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:cd4f696dd0a357341fb46068a2188905ed3a3da1bd156a39e8c209d308a4c321",
      "size": 566,
      "annotations": {
        "vnd.docker.reference.digest": "sha256:385c5ec07e72af7f5acb6173b5dc704f05ffc33ec14c97ed961d62d2a5cceba7",
        "vnd.docker.reference.type": "attestation-manifest"
      },
      "platform": {
        "architecture": "unknown",
        "os": "unknown"
      }
    },
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:6d4e8c5beb404295feae71160db4d331e2c0a01a7e43b38bb4e976937721e531",
      "size": 566,
      "annotations": {
        "vnd.docker.reference.digest": "sha256:4c48a7c712ff3446d982c7131ecf87b4e17f25495b8ac27789e5aedfd28762ae",
        "vnd.docker.reference.type": "attestation-manifest"
      },
      "platform": {
        "architecture": "unknown",
        "os": "unknown"
      }
    }
  ]
}
```

</details>
<hr/>

### Image manifest:
`curl -X GET -H 'Accept: application/vnd.oci.image.manifest.v1+json' https://<repo>/v2/gorymoon/sample/manifests/sha256:385c5ec07e72af7f5acb6173b5dc704f05ffc33ec14c97ed961d62d2a5cceba7`
<details>
<summary>Click to show Image Manifest</summary>

```json
{
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "schemaVersion": 2,
  "config": {
    "mediaType": "application/vnd.oci.image.config.v1+json",
    "digest": "sha256:66c190bfa8bd821052e95f32c632df35ab6ba68ab7449d0ae5e37241330d1c1e",
    "size": 7333
  },
  "layers": [
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:9e3ea8720c6de96cc9ad544dddc695a3ab73f5581c5d954e0504cc4f80fb5e5c",
      "size": 31403516
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:bf36b64666794f28ea5c3d4d75add149c04e849342e3d45ca31aac9cf4715dd1",
      "size": 25584344
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:15a97cf85bb88997d139f86b2be23f99175d51d7e45a4c4b04ec0cbdbb56a63b",
      "size": 626
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:9c2d6be5a61d1ad44be8e5e93a5800572cff95601147c45eaa9ecf0d4cb66f83",
      "size": 958
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:6b7e4a5c7c7ad54c76bc4861f476f3b70978beede9e752015202dd223383602b",
      "size": 773
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:8db4caa19df89c606d39076b27fe163e1f37516f889ff5bfee1fce03056bb6b0",
      "size": 1403
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:c8d0cae51d7e7f7ab3c01478ad777841448c93f6cbd0f217edbec0153d179361",
      "size": 440
    }
  ]
}
```

</details>
<hr/>

### Image config:
`curl -X GET https://<repo>/v2/gorymoon/sample/blobs/sha256:66c190bfa8bd821052e95f32c632df35ab6ba68ab7449d0ae5e37241330d1c1e`
<details>
<summary>Click to show Config</summary>

```json
{
  "architecture": "amd64",
  "config": {
    "ExposedPorts": {
      "80/tcp": {}
    },
    "Env": [
      "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
      "NGINX_VERSION=1.23.4",
      "NJS_VERSION=0.7.11",
      "PKG_RELEASE=1~bullseye"
    ],
    "Entrypoint": [
      "/docker-entrypoint.sh"
    ],
    "Cmd": [
      "nginx",
      "-g",
      "daemon off;"
    ],
    "Labels": {
      "maintainer": "NGINX Docker Maintainers <docker-maint@nginx.com>"
    },
    "StopSignal": "SIGQUIT",
    "OnBuild": null
  },
  "created": "2023-05-16T22:52:44.798347171Z",
  "history": [
    {
      "created": "2023-05-02T23:46:59.831699727Z",
      "created_by": "/bin/sh -c #(nop) ADD file:a2378c1b12e95db69e24b9d347441678c6f23239292cce3c822b1524992b6ec4 in / "
    },
    {
      "created": "2023-05-02T23:47:00.1128963Z",
      "created_by": "/bin/sh -c #(nop)  CMD [\"bash\"]",
      "empty_layer": true
    },
    {
      "created": "2023-05-03T19:50:47.862144405Z",
      "created_by": "/bin/sh -c #(nop)  LABEL maintainer=NGINX Docker Maintainers <docker-maint@nginx.com>",
      "empty_layer": true
    },
    {
      "created": "2023-05-03T19:50:47.939315234Z",
      "created_by": "/bin/sh -c #(nop)  ENV NGINX_VERSION=1.23.4",
      "empty_layer": true
    },
    {
      "created": "2023-05-03T19:50:48.020227271Z",
      "created_by": "/bin/sh -c #(nop)  ENV NJS_VERSION=0.7.11",
      "empty_layer": true
    },
    {
      "created": "2023-05-03T19:50:48.097124251Z",
      "created_by": "/bin/sh -c #(nop)  ENV PKG_RELEASE=1~bullseye",
      "empty_layer": true
    },
    {
      "created": "2023-05-03T19:51:05.443923646Z",
      "created_by": "/bin/sh -c set -x    ..."
    },
    {
      "created": "2023-05-03T19:51:05.651432117Z",
      "created_by": "/bin/sh -c #(nop) COPY file:7b307b62e82255f040c9812421a30090bf9abf3685f27b02d77fcca99f997911 in / "
    },
    {
      "created": "2023-05-03T19:51:05.73780747Z",
      "created_by": "/bin/sh -c #(nop) COPY file:5c18272734349488bd0c94ec8d382c872c1a0a435cca13bd4671353d6021d2cb in /docker-entrypoint.d "
    },
    {
      "created": "2023-05-03T19:51:05.821337624Z",
      "created_by": "/bin/sh -c #(nop) COPY file:abbcbf84dc17ee4454b6b2e3cf914be88e02cf84d344ec45a5b31235379d722a in /docker-entrypoint.d "
    },
    {
      "created": "2023-05-03T19:51:05.906294486Z",
      "created_by": "/bin/sh -c #(nop) COPY file:e57eef017a414ca793499729d80a7b9075790c9a804f930f1417e56d506970cf in /docker-entrypoint.d "
    },
    {
      "created": "2023-05-03T19:51:05.979700896Z",
      "created_by": "/bin/sh -c #(nop)  ENTRYPOINT [\"/docker-entrypoint.sh\"]",
      "empty_layer": true
    },
    {
      "created": "2023-05-03T19:51:06.058893465Z",
      "created_by": "/bin/sh -c #(nop)  EXPOSE 80",
      "empty_layer": true
    },
    {
      "created": "2023-05-03T19:51:06.138099974Z",
      "created_by": "/bin/sh -c #(nop)  STOPSIGNAL SIGQUIT",
      "empty_layer": true
    },
    {
      "created": "2023-05-03T19:51:06.214253945Z",
      "created_by": "/bin/sh -c #(nop)  CMD [\"nginx\" \"-g\" \"daemon off;\"]",
      "empty_layer": true
    },
    {
      "created": "2023-05-16T22:52:44.798347171Z",
      "created_by": "COPY index.html /usr/share/nginx/html # buildkit",
      "comment": "buildkit.dockerfile.v0"
    }
  ],
  "moby.buildkit.buildinfo.v1": "eyJmcm9udGVuZCI6ImRvY2tlcmZpbGUudjAiLCJzb3VyY2VzIjpbeyJ0eXBlIjoiZG9ja2VyLWltYWdlIiwicmVmIjoiZG9ja2VyLmlvL2xpYnJhcnkvbmdpbng6bGF0ZXN0IiwicGluIjoic2hhMjU2OjQ4MDg2OGU4YzhjNzk3Nzk0MjU3ZTJhYmQ4OGQwZjlhODgwOWIyZmU5NTZjYmZiYzA1ZGNjMGJjYTFmN2NkNDMifV19",
  "os": "linux",
  "rootfs": {
    "type": "layers",
    "diff_ids": [
      "sha256:8553b91047dad45bedc292812586f1621e0a464a09a7a7c2ce6ac5f8ba2535d7",
      "sha256:a29cc9587af6488ae0cbb962ecbe023d347908cc62ca5d715af06e54ccaa9e36",
      "sha256:6bc8ae8fb3cf0909b3d9c2e74f6cabe16e6a2322c52cec76fbaecaef47006f1d",
      "sha256:5684be535bf11cb9ad1a57b51085f36d84ae8361eabc2b4c2ba9a83e8b084b20",
      "sha256:93ee76f39c974e4f819e632149c002d6f509aadc5995ec6523a96b337751c8ed",
      "sha256:1040838fe30e6f26d31bde96c514f47ee4bf727b3f1c3c7b045ea3891c1c2150",
      "sha256:351bdc7f9386666decb4bc645db211ccdfe9b2cf0fa9eeb1a0e0fd0fb60a22a2"
    ]
  }
}
```

</details>

**Comment by Joxit** [comment]
Hi there and thank you again for all your inputs :smile: 

I pushed on my server 3 images, one with buildkit (`application/vnd.oci.image.layer.v1.tar+gzip`), one with buildx with `--provenance true` (`application/vnd.oci.image.manifest.v1+json`) and the last one wit buildx and `--provenance false` (`application/vnd.docker.distribution.manifest.v2+json`). That way I check if everything is ok with all the cases.

I think I almost fixed this issue, I need to create some tests now. Stay tunes :smile:

**Comment by Joxit** [comment]
Done in the latest commit, should be available with the tag `main` right now and will be available in [2.5.0](https://github.com/Joxit/docker-registry-ui/milestone/6)
